stablehlo/dialect/TypeInference.cpp

```diff
@@ -1443,8 +1443,83 @@ LogicalResult inferCholeskyOp(
     return emitOptionalError(
         location, "minor dimensions of 'a' must have equal size, got shape ",
         aShape, ".");
-  inferredReturnShapes.emplace_back(aRankedType.getShape(),
-                                    aRankedType.getElementType());
```

Comment:
Thank you for your work going into detail of all the possible cases for inferring the most specific dims for the result type! This was some meticulous work that was important in influencing the overall guidelines that we've worked out for bounded dynamism in shape functions.